Admin Dashboard:View, Del/Acti, Roles, Logs

AilerssiLP opened the pull request:
Admin Dashboard: View users with some filters, Delete users/Activate and Deactivate Users, Roles - Users, Mods, Admins , Logs - of activities like creation , comments and likes...

Commit:
Admin Dashboard:View,Del/Acti, Roles, Logs

vishal-kumar3 commented:
Really nice work @AilerssiLP, but there are few changes i want you to do may be a lot...
**It was my fault that i didn't clarified the UI.**
* Checkout this comment by me for ui reference:- https://github.com/vishal-kumar3/WriteFlow/issues/13#issuecomment-2392950810
* First of all git pull all the changes.
* Work on the admin folder that i created.
* For UI reference you can checkout pocketbase.io.
* And try breaking down the UI into components it will be easy to understand and maintainable in future.

And if you have any other questions or have any confusion then reach me at discord:- https://discord.com/users/954279073234956308

## UI Changes
![Screenshot from 2024-10-09 14-38-19](https://github.com/user-attachments/assets/96e77642-0ba3-45a5-b214-44bc7b501b18)
![Screenshot from 2024-10-09 14-38-37](https://github.com/user-attachments/assets/908180c3-fcda-4f2c-b71f-f24e3a9066d5)
![image](https://github.com/user-attachments/assets/48589a54-42ec-47e8-84e1-ddbdbd7cf5df)

## Use build command to ensure there isn't any bugs
* Fix these bugs
![image](https://github.com/user-attachments/assets/963f0f8a-5741-45ed-979b-cd91b7a397c4)
![image](https://github.com/user-attachments/assets/706e6303-d360-4465-b318-a0a63a2c4a0a)

## Folder structure
* Create separate folder for admin/user in the components folder and keeps all the related files there.
![image](https://github.com/user-attachments/assets/7d740810-756f-4dfe-83b1-ee1c7e9a9909)
![image](https://github.com/user-attachments/assets/2ba4d175-54c0-45c4-8c28-08d8f5ec5d82)
![image](https://github.com/user-attachments/assets/cdcc2fc5-178a-4775-927d-acee1bf4fd75)

vishal-kumar3 commented:
## You can use const session = await auth(); to get user id ( session.user.id ) [ instead of sessionProvider ] .This way all you can use server components
![image](https://github.com/user-attachments/assets/eccc8e72-0ebc-4fd6-a170-adf71dc782ec)

## use prisma on the server component to fetch data and server actions or apis to make changes to database
![image](https://github.com/user-attachments/assets/2d8b3e70-2941-4cd1-8edc-88c11260bdb9)
 ->  Example:- 
![image](https://github.com/user-attachments/assets/fef96406-852f-493d-8ad9-daafcdbcc6a0)

vishal-kumar3 commented:
And remove the files like .env.local, yarn.lock, package-lock.json, bun.lockb when you commit since they are not needed at myside

AilerssiLP commented:
Oki doki, i will try to remake it all as y suggested. 
Thank you for the tips! 
I will try to do most of it today/tommorrow depending on, if somethink gives me trouble.

Commit:
Checkout this comment by me for ui reference:- [Feature Request]: Create a Admin dashboard page for content and report management. #13 (comment) - done similarly .

First of all git pull all the changes. - it should be up to date .

For UI reference you can checkout pocketbase.io. - checked it out and tried to do it.

And try breaking down the UI into components it will be easy to understand and maintainable in future. - did as best as i could i think the rest would be better for you to do.

Fix bugs - fixed bugs related to my code.

Folder structure - i tried to do it as you asked but i think you should rearange it as y prefer in the end.

replaced usereffect and used prisma except for logs had issues so i left it as is....

removing files .env.local, yarn.lock,package,bun - done
